pallets/rc-migrator/src/lib.rs

```diff
@@ -812,16 +823,46 @@ pub mod pallet {
 		pub fn resend_xcm(origin: OriginFor<T>, query_id: u64) -> DispatchResultWithPostInfo {
 			<T as Config>::ManagerOrigin::ensure_origin(origin)?;
 
-			let xcm = PendingXcmMessages::<T>::get(query_id).ok_or(Error::<T>::QueryNotFound)?;
+			let message_hash =
+				PendingXcmQueries::<T>::get(query_id).ok_or(Error::<T>::QueryNotFound)?;
+			let xcm =
+				PendingXcmMessages::<T>::get(message_hash).ok_or(Error::<T>::QueryNotFound)?;
+
+			let asset_hub_location = Location::new(0, Parachain(1000));
+			let receive_notification_call =
+				Call::<T>::receive_query_response { query_id: 0, response: Default::default() };
 
-			if let Err(err) = send_xcm::<T::SendXcm>(Location::new(0, Parachain(1000)), xcm) {
+			let new_query_id = pallet_xcm::Pallet::<T>::new_notify_query(
+				asset_hub_location.clone(),
+				<T as Config>::RuntimeCall::from(receive_notification_call),
+				frame_system::Pallet::<T>::block_number() + T::XcmResponseTimeout::get(),
+				Location::here(),
+			);
+
+			let xcm_with_report = {
+				let mut xcm = xcm.clone();
+				xcm.inner_mut().push(SetAppendix(Xcm(vec![ReportTransactStatus(
```

Comment:
iirc, if you append `SetAppendix` at the end as the last instruction and any of the previous instructions fail, then the appendix won't be triggered, and `ReportTransactStatus` won't send anything to `destination: Location::parent()`, right?

I think that if you also want `SetAppendix` to be triggered in case of an error, you need to append it after `UnpaidExecution` or at least before `Transact`.

But I still don't fully know the complete flow involving sending, resend_xcm, and ReportTransactStatus handling - maybe the order of `SetAppendix` isn't actually an issue.

Reply:
It make sure it executed after the rest of the message is processed and also makes sure it will be executed even for some reason the execution of the rest of the messages exited with an error. 
Do you see any problem with using `SetAppendix`?

Reply:
if its ultra nit (= I would not touch it, just not worth adjusting the code everywhere, running tests again, etc.